rdt-app/src/main/java/io/ona/rdt_app/util/RDTJsonFormUtils.java

```diff
@@ -148,26 +149,37 @@ public void launchForm(Activity activity, String formName) throws JSONException
         launchForm(activity, formName, null);
     }
 
-    public void launchForm(Activity activity, String formName, String entityId) throws JSONException {
+    public void launchForm(Activity activity, String formName, Patient patient) throws JSONException {
         try {
             JSONObject formJsonObject = getFormJsonObject(formName, activity);
             String rdtId = Constants.Form.RDT_TEST_FORM.equals(formName) ? UUID.randomUUID().toString().substring(0, 5) : "";
-            prePopulateFormFields(formJsonObject, entityId, rdtId);
+            prePopulateFormFields(formJsonObject, patient, rdtId);
             startJsonForm(formJsonObject, activity, REQUEST_CODE_GET_JSON);
         } catch (JsonFormMissingStepCountException e) {
             Log.e(TAG, e.getStackTrace().toString());
         }
     }
 
-    public void prePopulateFormFields(JSONObject jsonForm, String entityId, String rdtId) throws JSONException, JsonFormMissingStepCountException{
-        jsonForm.put(ENTITY_ID, entityId);
+    public void prePopulateFormFields(JSONObject jsonForm, Patient patient, String rdtId) throws JSONException, JsonFormMissingStepCountException{
```

Comment:
Is it possible to track if fields, values have been set so that the loop can be broken, this will save some CPU cycles

Reply:
Done.